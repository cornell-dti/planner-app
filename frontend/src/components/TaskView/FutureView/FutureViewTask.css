.Task {
  position: relative;
  color: white;
  cursor: pointer;
  margin: 0;
  padding: 0;
  font-weight: 500;
}

.TaskMainWrapper {
  display: flex;
  padding: 3px;
  align-items: center;
  font-size: 12px;
  font-weight: 300;
}

.SubTask {
  display: flex;
  padding: 3px;
  align-items: center;
  color: #666666;
  background-color: white;
  font-size: 12px;
  font-weight: 300;
}

.TaskCheckBox {
  margin-left: 8px;
  margin-right: 5px;
}

.TaskCheckBoxPlaceHolder {
  margin-left: 8px;
  width: 16px;
  margin-right: 5px;
}

.TaskIcon {
  flex: none;
  cursor: pointer;
  margin: 0 4px 0 0 !important;
  height: 16px !important;
}

.TaskText {
  flex: 1 1 auto;
  margin-left: 3px;
  margin-right: 3px;
  font-size: 12px;
  height: 19px;
  font-weight: normal;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap; 
}
