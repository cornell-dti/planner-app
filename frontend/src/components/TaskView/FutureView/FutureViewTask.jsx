// @flow strict

import React from 'react';
import type { ComponentType, Node } from 'react';
import { Icon } from 'semantic-ui-react';
import Delete from '../../../assets/svgs/X.svg';
import PinFilled from '../../../assets/svgs/pin-2-dark-filled.svg';
import PinOutline from '../../../assets/svgs/pin-2-dark.svg';
import styles from './FutureViewTask.css';
import FutureViewSubTask from './FutureViewSubTask';
import FloatingTaskEditor from '../../Util/TaskEditors/FloatingTaskEditor';
import type { SubTask, Task } from '../../../store/store-types';
import CheckBox from '../../UI/CheckBox';
import type { FloatingPosition } from '../../Util/TaskEditors/task-editors-types';
import { getTodayAtZeroAM } from '../../../util/datetime-util';
import OverdueAlert from '../../UI/OverdueAlert';
import windowSizeConnect from '../../Util/Responsive/WindowSizeConsumer';
import type { WindowSize } from '../../Util/Responsive/window-size-context';
import { nDaysViewHeaderHeight, otherViewsHeightHeader } from './future-view-css-props';
import { error } from '../../../util/general-util';
import { editMainTask, removeTask } from '../../../firebase/actions';

type OwnProps = {|
  +originalTask: Task;
  +filteredTask: Task;
  +taskColor: string;
  +inNDaysView: boolean;
  +isInMainList: boolean;
  +taskEditorPosition: FloatingPosition;
|};

type Props = {|
  ...OwnProps;
  // Subscribed window size
  +windowSize: WindowSize;
|};

type State = {|
  +overdueAlertPosition: ?{| +top: number; +right: number; |};
|};

/**
 * The component used to render one task in backlog day.
 */
class FutureViewTask extends React.PureComponent<Props, State> {
  state: State = { overdueAlertPosition: null };

  /**
   * Get an onClickHandler when the element is clicked.
   * This methods ensure that only clicking on task text counts.
   *
   * @param {function(): void} opener the opener passed by the floating task editor.
   * @return {function} the onClick handler.
   */
  getOnClickHandler = (opener: () => void) => (event: SyntheticEvent<HTMLElement>): void => {
    if (event.target instanceof HTMLElement) {
      const elem: HTMLElement = event.target;
      // only accept click on text.
      if (elem.className === styles.TaskText) {
        opener();
      }
    }
  };

  renderCheckBox = (): Node => {
    const { filteredTask: { id, complete } } = this.props;
    const onChange = () => {
      editMainTask(id, { complete: !complete });
    };
    return <CheckBox className={styles.TaskCheckBox} checked={complete} onChange={onChange} />;
  };

  renderTaskName = (): Node => {
    const { filteredTask: { name, complete } } = this.props;
    const tagStyle = complete ? { textDecoration: 'line-through' } : {};
    return <span className={styles.TaskText} style={tagStyle}>{name}</span>;
  };

  renderRemoveTaskIcon = (): Node => {
    const { filteredTask } = this.props;
    const handler = () => {
      removeTask(filteredTask);
    };
    // onClick={handler}
    return <span style={{ backgroundImage: `url(${Delete})` }} className={styles.TaskIcon} />;
  };

  renderBookmarkIcon = (): Node => {
    const { filteredTask: { id, inFocus } } = this.props;
    const iconName = inFocus ? PinFilled : PinOutline;
    const handler = () => editMainTask(id, { inFocus: !inFocus });
    return <span style={{ backgroundImage: `url(${iconName})` }} className={styles.TaskIcon} onClick={handler} />;
  };

  renderMainTaskInfo = (simplified: boolean = false): Node => {
    const { taskColor, isInMainList } = this.props;
    if (simplified && isInMainList) {
      const style = { backgroundColor: taskColor, height: '25px' };
      return <div className={styles.TaskMainWrapper} style={style} />;
    }
    return (
      <div className={styles.TaskMainWrapper} style={{ backgroundColor: taskColor }}>
        {this.renderCheckBox()}
        {this.renderTaskName()}
        {this.renderBookmarkIcon()}
        {this.renderRemoveTaskIcon()}
      </div>
    );
  };

  renderSubTasks = (): Node => {
    const { filteredTask: { complete, subtasks } } = this.props;
    return subtasks.map((subTask: SubTask) => (
      <FutureViewSubTask
        key={subTask.id}
        mainTaskCompleted={complete}
        {...subTask}
      />
    ));
  };

  render(): Node {
    const { inNDaysView, taskEditorPosition, originalTask } = this.props;
    const { date, complete } = originalTask;
    const { overdueAlertPosition } = this.state;
    const overdueComponentOpt = overdueAlertPosition && (
      <OverdueAlert absolutePosition={overdueAlertPosition} />
    );
    const refHandler = (divElement: HTMLDivElement | null) => {
      if (divElement != null) {
        const isOverdue = date < getTodayAtZeroAM() && !complete;
        if (isOverdue && overdueAlertPosition == null) {
          const { top } = divElement.getBoundingClientRect();
          const parent = divElement.parentElement ?? error('Corrupted DOM!');
          const parentRect = parent.getBoundingClientRect();
          const headerHeight = inNDaysView ? nDaysViewHeaderHeight : otherViewsHeightHeader;
          this.setState({
            overdueAlertPosition: {
              top: top - parentRect.top + headerHeight - 3,
              right: -5,
            },
          });
        }
      }
    };
    // Construct the trigger for the floating task editor.
    const trigger = (opened: boolean, opener: () => void): Node => {
      const onClickHandler = this.getOnClickHandler(opener);
      const style = opened ? { zIndex: 8 } : {};
      const { windowSize: { width } } = this.props;
      const mainTasks = inNDaysView
        ? this.renderMainTaskInfo()
        : this.renderMainTaskInfo(width <= 768);
      const subtasks = inNDaysView ? this.renderSubTasks() : null;
      return (
        <div
          role="presentation"
          className={styles.Task}
          style={style}
          onClick={onClickHandler}
          ref={refHandler}
        >
          {overdueComponentOpt}
          {mainTasks}
          {subtasks}
        </div>
      );
    };
    return (
      <FloatingTaskEditor
        position={taskEditorPosition}
        initialTask={originalTask}
        trigger={trigger}
      />
    );
  }
}

const Connected: ComponentType<OwnProps> = windowSizeConnect(FutureViewTask);
export default Connected;
