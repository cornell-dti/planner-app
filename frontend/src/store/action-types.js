// @flow strict

import type {
  PartialMainTask, PartialSubTask, SubTask, Tag, Task, Course,
} from './store-types';
import type { TaskDiff } from '../util/task-util';

/*
 * --------------------------------------------------------------------------------
 * Part 1: Task Actions
 * --------------------------------------------------------------------------------
 */

export type AddNewTaskAction = {| type: 'ADD_NEW_TASK'; +task: Task; |};
export type AddNewSubTaskAction = {|
  type: 'ADD_NEW_SUBTASK'; +taskId: number; +subTask: SubTask;
|};

export type EditTaskAction = {| type: 'EDIT_TASK'; +task: Task; +diff: TaskDiff; |};
export type EditMainTaskAction = {|
  type: 'EDIT_MAIN_TASK'; +taskId: number; +partialMainTask: PartialMainTask;
|};
export type EditSubTaskAction = {|
  type: 'EDIT_SUB_TASK'; +taskId: number; +subtaskId: number; +partialSubTask: PartialSubTask;
|};

export type RemoveTaskAction = {| type: 'REMOVE_TASK'; +taskId: number; |};
export type RemoveSubTaskAction = {|
  type: 'REMOVE_SUBTASK'; +taskId: number; +subtaskId: number;
|};

export type ImportCourseExamsAction = {| type: 'IMPORT_COURSE_EXAMS' |};

type TaskAction =
  | AddNewTaskAction
  | AddNewSubTaskAction
  | EditTaskAction
  | EditMainTaskAction
  | EditSubTaskAction
  | RemoveTaskAction
  | RemoveSubTaskAction
  | ImportCourseExamsAction;

/*
 * --------------------------------------------------------------------------------
 * Part 2: Color Config Actions
 * --------------------------------------------------------------------------------
 */

export type AddTagAction = {| type: 'ADD_TAG'; +tag: Tag; |};
export type EditTagAction = {| type: 'EDIT_TAG'; +tag: Tag; |};
export type RemoveTagAction = {| type: 'REMOVE_TAG'; +tagId: number; |};

export type TagAction = AddTagAction | EditTagAction | RemoveTagAction;

/*
 * --------------------------------------------------------------------------------
 * Part 3: Undo Actions
 * --------------------------------------------------------------------------------
 */

export type UndoAddTaskAction = {| type: 'UNDO_ADD_TASK' |};
export type ClearUndoAddTaskAction = {| type: 'CLEAR_UNDO_ADD_TASK' |};
export type UndoDeleteTaskAction = {| type: 'UNDO_DELETE_TASK' |};
export type ClearUndoDeleteTaskAction = {| type: 'CLEAR_UNDO_DELETE_TASK' |};

export type UndoAction =
  | UndoAddTaskAction
  | ClearUndoAddTaskAction
  | UndoDeleteTaskAction
  | ClearUndoDeleteTaskAction;

/*
 * --------------------------------------------------------------------------------
 * Part 4: Backend Patch Actions
 * --------------------------------------------------------------------------------
 */

export type BackendPatchNewTagAction = {|
  type: 'BACKEND_PATCH_NEW_TAG'; +tempId: number; +backendId: number;
|};
export type BackendPatchNewTaskAction = {|
  type: 'BACKEND_PATCH_NEW_TASK'; +tempId: number; +backendTask: Task;
|};
export type BackendPatchBatchNewTasksAction = {|
  type: 'BACKEND_PATCH_BATCH_NEW_TASKS'; +tempIds: number[]; +backendTasks: Task[];
|};
export type BackendPatchNewSubTaskAction = {|
  type: 'BACKEND_PATCH_NEW_SUBTASK';
  +taskId: number;
  +tempSubTaskId: number;
  +backendSubTaskId: number;
|};
export type BackendPatchExistingTaskAction = {|
  type: 'BACKEND_PATCH_EXISTING_TASK'; +task: Task;
|};
export type BackendPatchLoadedDataAction = {|
  type: 'BACKEND_PATCH_LOADED_DATA'; +tags: Tag[]; +tasks: Task[]; +courses: Map<number, Course[]>;
|};

/**
 * All the tasks that lets the backend patch the store with all the latest values.
 */
export type BackendPatchAction =
  | BackendPatchNewTagAction
  | BackendPatchNewTaskAction
  | BackendPatchBatchNewTasksAction
  | BackendPatchNewSubTaskAction
  | BackendPatchExistingTaskAction
  | BackendPatchLoadedDataAction

/*
 * --------------------------------------------------------------------------------
 * Part 5: All Actions
 * --------------------------------------------------------------------------------
 */

export type Action = TagAction | TaskAction | UndoAction | BackendPatchAction;
