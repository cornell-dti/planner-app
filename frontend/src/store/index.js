// @flow strict

import { createStore } from 'redux';
import type { Store } from 'redux';
import rootReducer from './reducers';
import type { State } from './store-types';
import type { Action } from './action-types';

const store: Store<State, $Subtype<Action>> = createStore(rootReducer);
export default store;
