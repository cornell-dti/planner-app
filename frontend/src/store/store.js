// @flow strict

import { createStore } from 'redux';
import type { Store } from 'redux';
import rootReducer from './reducers';
import type { State } from './store-types';
import type { Action } from './action-types';
import type { AppUser } from '../util/firebase-util';
import { error } from '../util/general-util';

export type GlobalStore = Store<State, $Subtype<Action>>;

const store: GlobalStore = createStore(rootReducer);
let appUser: AppUser | null = null;

/**
 * Initialize the app and record a global user and returns the initialized store.
 *
 * @param {AppUser} user the user of the app.
 * @return {GlobalStore} the newly initialized store.
 */
export function initializeApp(user: AppUser): GlobalStore {
  appUser = user;
  return store;
}

/**
 * Returns the global app user.
 *
 * @return {AppUser} the global app user.
 */
export const getAppUser = (): AppUser => appUser ?? error('App is not initialized.');

/**
 * Dispatch an action and returns the dispatched action.
 *
 * @param {Action} action the action to dispatch.
 * @return {Action} the action to dispatch.
 */
export const dispatchAction = (action: Action): Action => store.dispatch(action);
