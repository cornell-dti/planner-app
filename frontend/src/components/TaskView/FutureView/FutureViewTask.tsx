import React, { KeyboardEvent, ReactElement, SyntheticEvent, ReactNode } from 'react';
import { connect } from 'react-redux';
import styles from './FutureViewTask.module.css';
import FutureViewSubTask from './FutureViewSubTask';
import FloatingTaskEditor from '../../Util/TaskEditors/FloatingTaskEditor';
import { State, SubTask, Task } from '../../../store/store-types';
import CheckBox from '../../UI/CheckBox';
import { FloatingPosition, CalendarPosition } from '../../Util/TaskEditors/editors-types';
import { getTodayAtZeroAM } from '../../../util/datetime-util';
import OverdueAlert from '../../UI/OverdueAlert';
import { editMainTask, removeTask } from '../../../firebase/actions';
import { useMappedWindowSize } from '../../../hooks/window-size-hook';
import { NONE_TAG } from '../../../util/tag-util';
import SamwiseIcon from '../../UI/SamwiseIcon';

type CompoundTask = {
  readonly original: Task;
  readonly filteredSubTasks: SubTask[];
  readonly color: string;
};

type OwnProps = {
  readonly taskId: string;
  readonly inNDaysView: boolean;
  readonly taskEditorPosition: FloatingPosition;
  readonly calendarPosition: CalendarPosition;
  readonly doesShowCompletedTasks: boolean;
  readonly isInMainList: boolean;
};

type Props = OwnProps & {
  readonly compoundTask: CompoundTask | null;
};

/**
 * The component used to render one task in backlog day.
 */
function FutureViewTask(
  {
    compoundTask, inNDaysView, taskEditorPosition, isInMainList, calendarPosition,
  }: Props,
): ReactElement | null {
  const isSmallScreen = useMappedWindowSize(({ width }) => width <= 768);

  if (compoundTask === null) {
    return null;
  }
  const { original, filteredSubTasks, color } = compoundTask;

  /**
   * Get an onClickHandler when the element is clicked.
   * This methods ensure that only clicking on task text counts.
   *
   * @param opener the opener passed by the floating task editor.
   * @return the onClick handler.
   */
  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
  const getOnClickHandler = (opener: () => void) => (event?: SyntheticEvent<HTMLElement>): void => {
    if (event == null) {
      opener();
      return;
    }
    if (event.target instanceof HTMLElement) {
      const elem: HTMLElement = event.target;
      // only accept click on text.
      if (elem.className === styles.TaskText) {
        opener();
      }
    }
  };

  const TaskCheckBox = (): ReactElement => {
    const { id, complete } = original;
    const onChange = (): void => editMainTask(id, { complete: !complete });
    return <CheckBox className={styles.TaskCheckBox} checked={complete} onChange={onChange} />;
  };
  const TaskName = (): ReactElement => {
    const { name, complete } = original;
    const tagStyle = complete ? { textDecoration: 'line-through' } : {};
    return <span className={styles.TaskText} style={tagStyle}>{name}</span>;
  };

  const RemoveTaskIcon = (): ReactElement => {
    const handler = (): void => removeTask(original);
    return <SamwiseIcon iconName="x-light" className={styles.TaskIcon} onClick={handler} />;
  };
  const PinIcon = (): ReactElement => {
    const { id, inFocus } = original;
    const handler = (): void => editMainTask(id, { inFocus: !inFocus });
    return (
      <SamwiseIcon
        iconName={inFocus ? 'pin-light-filled' : 'pin-light-outline'}
        className={styles.TaskIcon}
        onClick={handler}
      />
    );
  };

  const renderMainTaskInfo = (simplified: boolean = false): ReactElement => {
    if (simplified && isInMainList) {
      const style = { backgroundColor: color, height: '25px' };
      return <div className={styles.TaskMainWrapper} style={style} />;
    }
    return (
      <div className={styles.TaskMainWrapper} style={{ backgroundColor: color }}>
        <TaskCheckBox />
        <TaskName />
        <PinIcon />
        <RemoveTaskIcon />
      </div>
    );
  };

  const renderSubTasks = (): ReactNode => filteredSubTasks.map(s => (
    <FutureViewSubTask
      key={s.id}
      subTask={s}
      mainTaskId={original.id}
      mainTaskCompleted={original.complete}
    />
  ));

  const { date, complete } = original;
  const overdueComponentOpt = (date < getTodayAtZeroAM() && !complete) && (
    <OverdueAlert target="future-view-task" />
  );
  // Construct the trigger for the floating task editor.
  const trigger = (opened: boolean, opener: () => void): ReactElement => {
    const onClickHandler = getOnClickHandler(opener);
    const onSpaceHandler = (e: KeyboardEvent<HTMLDivElement>): void => {
      if (e.key === ' ') {
        onClickHandler();
      }
    };
    const style = opened ? { zIndex: 8 } : {};
    const mainTasks = inNDaysView
      ? renderMainTaskInfo()
      : renderMainTaskInfo(isSmallScreen);
    const subtasks = inNDaysView ? renderSubTasks() : null;
    return (
      <div
        role="button"
        tabIndex={0}
        className={styles.Task}
        style={style}
        onClick={onClickHandler}
        onKeyUp={onSpaceHandler}
      >
        {overdueComponentOpt}
        {mainTasks}
        {subtasks}
      </div>
    );
  };
  return (
    <FloatingTaskEditor
      position={taskEditorPosition}
      calendarPosition={calendarPosition}
      initialTask={original}
      trigger={trigger}
    />
  );
}

const getCompoundTask = (
  { tasks, subTasks, tags }: State, { taskId, doesShowCompletedTasks }: OwnProps,
): CompoundTask | null => {
  const original = tasks.get(taskId);
  if (original == null) {
    return null;
  }
  const { color } = tags.get(original.tag) || NONE_TAG;
  if (doesShowCompletedTasks) {
    let filteredSubTasks: SubTask[] = [];
    original.children.forEach((subTaskId) => {
      const s = subTasks.get(subTaskId);
      if (s != null) { filteredSubTasks.push(s); }
    });
    filteredSubTasks = filteredSubTasks.sort((a, b) => a.order - b.order);
    return { original, filteredSubTasks, color };
  }
  if (original.complete) {
    return null;
  }
  let filteredSubTasks: SubTask[] = [];
  original.children.forEach((subTaskId) => {
    const s = subTasks.get(subTaskId);
    if (s != null && !s.complete) { filteredSubTasks.push(s); }
  });
  filteredSubTasks = filteredSubTasks.sort((a, b) => a.order - b.order);
  return { original, filteredSubTasks, color };
};

const mapStateToProps = (
  state: State, ownProps: OwnProps,
): { readonly compoundTask: CompoundTask | null } => ({
  compoundTask: getCompoundTask(state, ownProps),
});

const Connected = connect(mapStateToProps)(FutureViewTask);
export default Connected;
