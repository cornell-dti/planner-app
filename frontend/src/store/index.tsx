import React, { ReactElement } from 'react';
import { Provider as ReactReduxProvider } from 'react-redux';
import { store, storeForTesting } from './store';

type Props = { readonly children: ReactElement };

export const Provider = ({ children }: Props): ReactElement => (
  <ReactReduxProvider store={store}>{children}</ReactReduxProvider>
);

export const ProviderForTesting = ({ children }: Props): ReactElement => (
  <ReactReduxProvider store={storeForTesting}>{children}</ReactReduxProvider>
);
