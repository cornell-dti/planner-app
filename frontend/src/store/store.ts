import { createStore, Store } from 'redux';
import rootReducer from './reducers';
import { State } from './store-types';
import { Action } from './action-types';
import { initialStateForTesting } from './state';

export type GlobalStore = Store<State, Action>;

/**
 * The redux store. It should only be given to the react-redux provider.
 */
export const store: GlobalStore = createStore(rootReducer);

/**
 * The redux store for testing.
 */
export const storeForTesting: GlobalStore = createStore(
  (initialState: State = initialStateForTesting, action: Action): State => rootReducer(
    initialState, action,
  ),
);

/**
 * Dispatch an action and returns the same dispatched action.
 */
export const dispatchAction = (action: Action): Action => store.dispatch(action);
