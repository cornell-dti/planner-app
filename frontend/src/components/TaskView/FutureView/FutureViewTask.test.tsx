import React from 'react';
import renderer from 'react-test-renderer';
import { ProviderForTesting } from 'store';
import FutureViewTask from './FutureViewTask';

const createTest = (
  inNDaysView: boolean, doesShowCompletedTasks: boolean, isInMainList: boolean,
): void => {
  const testName = `FutureViewTask(${inNDaysView}, ${doesShowCompletedTasks}, ${isInMainList}) matches snapshot.`;
  it(testName, () => {
    const tree = renderer.create(
      <ProviderForTesting>
        <FutureViewTask
          taskId="foo"
          containerDate="2030-01-01"
          inNDaysView={inNDaysView}
          taskEditorPosition="left"
          calendarPosition="top"
          doesShowCompletedTasks={doesShowCompletedTasks}
          isInMainList={isInMainList}
        />
      </ProviderForTesting>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

createTest(false, false, false);
createTest(false, false, true);
createTest(false, true, false);
createTest(false, true, true);
createTest(true, false, false);
createTest(true, false, true);
createTest(true, true, false);
createTest(true, true, true);
