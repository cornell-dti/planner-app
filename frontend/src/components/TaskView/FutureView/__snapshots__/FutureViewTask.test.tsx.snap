// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FutureViewTask(false, false, false) matches snapshot. 1`] = `null`;

exports[`FutureViewTask(false, false, true) matches snapshot. 1`] = `null`;

exports[`FutureViewTask(false, true, false) matches snapshot. 1`] = `
<div
  className="Task"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  style={Object {}}
  tabIndex={0}
>
  <div
    className="TaskMainWrapper"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    <span
      title="Reorder"
    >
      <svg
        alt="Reorder"
        className="TaskIcon"
        height="1em"
        onKeyUp={[Function]}
        tabIndex={0}
        title="Reorder"
        width="1em"
      >
        grabbers.svg
      </svg>
    </span>
    <label
      className="TaskCheckBox CheckBox"
    >
      <input
        defaultChecked={true}
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="checkbox"
      />
      <span
        className="CheckBoxCheckMark"
      />
    </label>
    <span
      className="TaskText"
      style={
        Object {
          "textDecoration": "line-through",
        }
      }
    >
      Foo
    </span>
    <span
      title="Pin to Focus"
    >
      <svg
        alt="Pin to Focus"
        className="TaskIcon"
        height="1em"
        onClick={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="Pin to Focus"
        width="1em"
      >
        pin-2-light-outline.svg
      </svg>
    </span>
    <span
      title="Delete"
    >
      <svg
        alt="Delete"
        className="TaskIcon"
        height="1em"
        onClick={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="Delete"
        width="1em"
      >
        XLight.svg
      </svg>
    </span>
  </div>
</div>
`;

exports[`FutureViewTask(false, true, true) matches snapshot. 1`] = `
<div
  className="Task"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  style={Object {}}
  tabIndex={0}
>
  <div
    className="TaskMainWrapper"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    <span
      title="Reorder"
    >
      <svg
        alt="Reorder"
        className="TaskIcon"
        height="1em"
        onKeyUp={[Function]}
        tabIndex={0}
        title="Reorder"
        width="1em"
      >
        grabbers.svg
      </svg>
    </span>
    <label
      className="TaskCheckBox CheckBox"
    >
      <input
        defaultChecked={true}
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="checkbox"
      />
      <span
        className="CheckBoxCheckMark"
      />
    </label>
    <span
      className="TaskText"
      style={
        Object {
          "textDecoration": "line-through",
        }
      }
    >
      Foo
    </span>
    <span
      title="Pin to Focus"
    >
      <svg
        alt="Pin to Focus"
        className="TaskIcon"
        height="1em"
        onClick={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="Pin to Focus"
        width="1em"
      >
        pin-2-light-outline.svg
      </svg>
    </span>
    <span
      title="Delete"
    >
      <svg
        alt="Delete"
        className="TaskIcon"
        height="1em"
        onClick={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="Delete"
        width="1em"
      >
        XLight.svg
      </svg>
    </span>
  </div>
</div>
`;

exports[`FutureViewTask(true, false, false) matches snapshot. 1`] = `null`;

exports[`FutureViewTask(true, false, true) matches snapshot. 1`] = `null`;

exports[`FutureViewTask(true, true, false) matches snapshot. 1`] = `
<div
  className="Task"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  style={Object {}}
  tabIndex={0}
>
  <div
    className="TaskMainWrapper"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    <span
      title="Reorder"
    >
      <svg
        alt="Reorder"
        className="TaskIcon"
        height="1em"
        onKeyUp={[Function]}
        tabIndex={0}
        title="Reorder"
        width="1em"
      >
        grabbers.svg
      </svg>
    </span>
    <label
      className="TaskCheckBox CheckBox"
    >
      <input
        defaultChecked={true}
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="checkbox"
      />
      <span
        className="CheckBoxCheckMark"
      />
    </label>
    <span
      className="TaskText"
      style={
        Object {
          "textDecoration": "line-through",
        }
      }
    >
      Foo
    </span>
    <span
      title="Pin to Focus"
    >
      <svg
        alt="Pin to Focus"
        className="TaskIcon"
        height="1em"
        onClick={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="Pin to Focus"
        width="1em"
      >
        pin-2-light-outline.svg
      </svg>
    </span>
    <span
      title="Delete"
    >
      <svg
        alt="Delete"
        className="TaskIcon"
        height="1em"
        onClick={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="Delete"
        width="1em"
      >
        XLight.svg
      </svg>
    </span>
  </div>
</div>
`;

exports[`FutureViewTask(true, true, true) matches snapshot. 1`] = `
<div
  className="Task"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  style={Object {}}
  tabIndex={0}
>
  <div
    className="TaskMainWrapper"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    <span
      title="Reorder"
    >
      <svg
        alt="Reorder"
        className="TaskIcon"
        height="1em"
        onKeyUp={[Function]}
        tabIndex={0}
        title="Reorder"
        width="1em"
      >
        grabbers.svg
      </svg>
    </span>
    <label
      className="TaskCheckBox CheckBox"
    >
      <input
        defaultChecked={true}
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        type="checkbox"
      />
      <span
        className="CheckBoxCheckMark"
      />
    </label>
    <span
      className="TaskText"
      style={
        Object {
          "textDecoration": "line-through",
        }
      }
    >
      Foo
    </span>
    <span
      title="Pin to Focus"
    >
      <svg
        alt="Pin to Focus"
        className="TaskIcon"
        height="1em"
        onClick={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="Pin to Focus"
        width="1em"
      >
        pin-2-light-outline.svg
      </svg>
    </span>
    <span
      title="Delete"
    >
      <svg
        alt="Delete"
        className="TaskIcon"
        height="1em"
        onClick={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="Delete"
        width="1em"
      >
        XLight.svg
      </svg>
    </span>
  </div>
</div>
`;
